docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/creating-vehicle-interface.md

```diff
@@ -0,0 +1,276 @@
+
+So, your `YOUR-OWN-VEHICLE-INTERFACE.hpp` header file should be like this:
+
+```c++
+...
+#include <autoware_auto_control_msgs/msg/ackermann_control_command.hpp>
+#include <autoware_auto_vehicle_msgs/msg/gear_report.hpp>
+...
+
+class <YOUR-OWN-INTERFACE> : public rclcpp::Node
+{
+public:
+    ...
+private:
+    ...
+    // from autoware
+    rclcpp::Subscription<autoware_auto_control_msgs::msg::AckermannControlCommand>::SharedPtr
+    control_cmd_sub_;
+    ...
+    // from vehicle
+    rclcpp::Publisher<autoware_auto_vehicle_msgs::msg::GearReport>::SharedPtr gear_status_pub_;
+    ...
+    // autoware command messages
+    ...
+    autoware_auto_control_msgs::msg::AckermannControlCommand::ConstSharedPtr control_cmd_ptr_;
+    ...
+    // callbacks
+    ...
+    void callback_control_cmd(
+    const autoware_auto_control_msgs::msg::AckermannControlCommand::ConstSharedPtr msg);
+    ...
+    void to_vehicle();
+    void from_vehicle();
+}
+```
+
+And your `YOUR-OWN-VEHICLE-INTERFACE.cpp` .cpp file should be like this:
+
+```c++
+#include <YOUR-OWN-VEHICLE-INTERFACE>/<YOUR-OWN-VEHICLE-INTERFACE>.hpp>
+...
+
+<YOUR-OWN-VEHICLE-INTERFACE>::<YOUR-OWN-VEHICLE-INTERFACE>()
+: Node("<YOUR-OWN-VEHICLE-INTERFACE>")
+{
+  ...
+  /* subscribers */
+  using std::placeholders::_1;
+  // from autoware
+  control_cmd_sub_ = create_subscription<autoware_auto_control_msgs::msg::AckermannControlCommand>(
+    "/control/command/control_cmd", 1, std::bind(&<YOUR-OWN-VEHICLE-INTERFACE>::callback_control_cmd, this, _1));
+  ...
+  // to autoware
+  gear_status_pub_ = create_publisher<autoware_auto_vehicle_msgs::msg::GearReport>(
+    "/vehicle/status/gear_status", rclcpp::QoS{1});
+  ...
+}
+
+void <YOUR-OWN-VEHICLE-INTERFACE>::callback_control_cmd(
+  const autoware_auto_control_msgs::msg::AckermannControlCommand::ConstSharedPtr msg)
+{
+  control_cmd_ptr_ = msg;
+}
+
+void <YOUR-OWN-VEHICLE-INTERFACE>::to_vehicle()
+{
+  ...
+  // you should implement this structure according to your own vehicle design
+  control_command_to_vehicle(control_cmd_ptr_);
+  ...
+}
+
+void <YOUR-OWN-VEHICLE-INTERFACE>::to_autoware()
+{
+  ...
+  // you should implement this structure according to your own vehicle design
+  autoware_auto_vehicle_msgs::msg::GearReport gear_report_msg;
+  convert_gear_status_to_autoware_msg(gear_report_msg);
+  gear_status_pub_->publish(gear_report_msg);
+  ...
+}
+
+```
+
+- Communication between the vehicle interface and your vehicle's control device
+
+  - If you are planning to drive by autoware with your vehicle, then your vehicle must satisfy some requirements:
+
+    | Type A                                                                                                                                         | Type B                                                                                                                                         |
```

Comment:
I believe Type A and Type B refers to the two types that are mentioned in `vehicle-interface.md` page.
Maybe we should consider either moving this table to `vehicle-interface` or move the explanation about the two types in `vehicle-interface.md` into this file.